### PR TITLE
[Merged by Bors] - Make the `iter_combinators` examples more pretty

### PR DESCRIPTION
**Pull request by DJMcNab**
# Objective

- More colours

## Solution

- Various changes
- Would have a video here, but don't have an easy way to record one before I go to sleep
- I intend to additionally change the distribution of the satellites, to be more uniform in space.

### CODE REVIEW

**Comment by mockersf** [comment]
https://user-images.githubusercontent.com/8672791/140589595-d7562887-70be-48d8-86f7-32e6f00760f3.mov

**Review by mockersf — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
bors try

File: examples/ecs/iter_combinations.rs [hunk 1]
```diff
@@ -101,29 +108,34 @@ fn generate_bodies(
...
             ..Default::default()
         })
         .insert(PointLight {
-            color: Color::ORANGE_RED,
-            ..Default::default()
-        });
+            color: Color::WHITE,
+            intensity: 400.,
+            range: 100.,
+            radius: star_radius,
```
Comment (alice-i-cecile) [inline examples/ecs/iter_combinations.rs:134]:
```suggestion
            radius: star_radius,
            ..Default::default(),
```
This should fix the compile failure the CI caught.

File: examples/ecs/iter_combinations.rs [hunk 2]
```diff
@@ -101,29 +108,35 @@ fn generate_bodies(
...
+            ..Default::default(),
```
Comment (mockersf) [inline examples/ecs/iter_combinations.rs:135]:
```suggestion
            ..Default::default()
```

**Comment by mockersf** [comment]
I tried running on rebased main, the light doesn't work anymore as it needs other components from the `PointLightBundle` 

Could you apply this change to your PR branch?
```diff
--- a/examples/ecs/iter_combinations.rs
+++ b/examples/ecs/iter_combinations.rs
@@ -127,14 +127,19 @@ fn generate_bodies(
             mass: Mass(500.0),
             ..Default::default()
         })
-        .insert(PointLight {
-            color: Color::WHITE,
-            intensity: 400.,
-            range: 100.,
-            radius: star_radius,
-            ..Default::default()
-        })
-        .insert(Star);
+        .insert(Star)
+        .with_children(|p| {
+            p.spawn_bundle(PointLightBundle {
+                point_light: PointLight {
+                    color: Color::WHITE,
+                    intensity: 400.0,
+                    range: 100.0,
+                    radius: star_radius,
+                    ..Default::default()
+                },
+                ..Default::default()
+            });
+        });
     commands.spawn_bundle(PerspectiveCameraBundle {
         transform: Transform::from_xyz(0.0, 10.5, -30.0).looking_at(Vec3::ZERO, Vec3::Y),
         ..Default::default()
```

**Review by mockersf — CHANGES_REQUESTED** [review]
I tried running on rebased main, the light doesn't work anymore as it needs other components from the `PointLightBundle` 

Could you apply this change to your PR branch?
```diff
--- a/examples/ecs/iter_combinations.rs
+++ b/examples/ecs/iter_combinations.rs
@@ -127,14 +127,19 @@ fn generate_bodies(
             mass: Mass(500.0),
             ..Default::default()
         })
-        .insert(PointLight {
-            color: Color::WHITE,
-            intensity: 400.,
-            range: 100.,
-            radius: star_radius,
-            ..Default::default()
-        })
-        .insert(Star);
+        .insert(Star)
+        .with_children(|p| {
+            p.spawn_bundle(PointLightBundle {
+                point_light: PointLight {
+                    color: Color::WHITE,
+                    intensity: 400.0,
+                    range: 100.0,
+                    radius: star_radius,
+                    ..Default::default()
+                },
+                ..Default::default()
+            });
+        });
     commands.spawn_bundle(PerspectiveCameraBundle {
         transform: Transform::from_xyz(0.0, 10.5, -30.0).looking_at(Vec3::ZERO, Vec3::Y),
         ..Default::default()
```

**Review by mockersf — APPROVED** [review]

**Pull request closed without merging**